### PR TITLE
RR-3 - Edit goal swagger spec

### PR DESCRIPTION
**Pull request by mattwills-moj-digital**
Swagger definition for a new endpoint to edit a Goal.

This endpoint is an "all or nothing" type affair - the whole Goal needs to be provided, or any missing data (e.g. a single Step) will be removed.

### CODE REVIEW

File: src/main/resources/static/openapi/EducationAndWorkPlanAPI.yml [hunk 1]
```diff
@@ -259,6 +291,65 @@ components:
...
+      required:
+        - goalReference
+        - title
+        - steps
+    UpdateStepRequest:
+      title: UpdateStepRequest
+      type: object
+      description: A request to create a Step within a Goal.
+      properties:
+        stepReference:
```
Comment (nathanrussell-moj-digital) [inline src/main/resources/static/openapi/EducationAndWorkPlanAPI.yml:332]: LGTM 👍 
My only thought/question would be whether `stepReference` should be a mandatory field or not ... 🤔 
I'm thinking about a future story as part of "editing a goal", where the screens allow for adding steps as part of "editing a goal". In this case we wont have a `stepReference` (its a new step); so I'm wondering if the API needs to support it in this way and therefore the implementation will need to work out what is new/what is edited/what is deleted as part of working out the diff

What do you think?

Comment (mattwills-moj-digital) [reply src/main/resources/static/openapi/EducationAndWorkPlanAPI.yml:332]: That's a good point, I'll change it 👍

File: src/testFixtures/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/resource/model/StepRequestBuilder.kt [hunk 2]
```diff
@@ -18,8 +19,21 @@ fun anotherValidCreateStepRequest(
...
-): CreateStepRequest =
-  CreateStepRequest(
+): StepRequest =
+  StepRequest(
+    title = title,
+    targetDateRange = targetDateRange,
+    sequenceNumber = sequenceNumber,
+  )
+
+fun aValidUpdateStepRequest(
```
Comment (mattwills-moj-digital) [inline src/testFixtures/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/resource/model/StepRequestBuilder.kt:29]: I've added this to show how we could use this builder for creating and updating a Step.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/resource/mapper/StepResourceMapper.kt [hunk 3]
```diff
@@ -13,9 +13,10 @@ import java.util.UUID
   ],
 )
 interface StepResourceMapper {
+  // TODO - RR-3 - this will need to change when creating vs updating a Step
```
Comment (nathanrussell-moj-digital) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/resource/mapper/StepResourceMapper.kt:16]: 👍

**Review by nathanrussell-moj-digital — APPROVED** [review]
LGTM 👍

**Pull request merged**